pkg/types/aws/validation/machinepool.go

```diff
@@ -47,6 +47,17 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("authentication"), p.EC2Metadata.Authentication, "must be either Required or Optional"))
 	}
 
+	allErrs = append(allErrs, validateSecurityGroups(platform, p, fldPath)...)
+
+	return allErrs
+}
+
+func validateSecurityGroups(platform *aws.Platform, p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(p.AdditionalSecurityGroupIDs) > 0 && len(platform.Subnets) <= 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("additionalSecurityGroupID"), "subnets must be provided for security groups"))
+	}
```

Comment:
Do we also want to validate that the IDs provided are in the format `sg-xxx`? Maybe not here but in the install-config validation?